02-microarray/pathway-analysis_microarray_02_ora_with_webgestaltr.Rmd

```diff
@@ -0,0 +1,262 @@
+We can do this with the `listIdType()` function; the first argument is the organism name.
+
+```{r}
+listIdType("hsapiens")
+```
+
+We can see that there is `ensembl_gene_id` listed above so the Ensembl IDs in the `Gene` column are compatible identifiers, and we do not need to convert to a different identifier.
+All refine.bio dataset downloads come with Ensembl IDs but if do find yourself needing to convert gene IDs, we'd encourage you to look at our [gene ID annotation examples](https://alexslemonade.github.io/refinebio-examples/02-microarray/gene-id-annotation_microarray_01_ensembl.html).
+
+The `WebGestaltR` function is a wrapper for the [WEB-based GEne SeT AnaLysis Toolkit (WebGestalt)](http://www.webgestalt.org/); note that WebGestalt has a new 2019 version) [@Wang2017].
+
+WebGestalt can perform multiple _types_ of pathway analysis. 
+Here we're using it for ORA and we will use [Gene Ontology (GO)](http://geneontology.org/docs/ontology-documentation/) biological processes as our source of gene sets.
+We can see all supported gene sets for humans with `listGeneSet()` function.
+We are using `head()` so we just get a preview and not the whole table printed out. 
+
+```{r}
+head(listGeneSet("hsapiens"))
+```
+
+The `WebGestaltR()` function will generate an HTML file with a report (path specified by the `outputDirectory` and `projectName` arguments) _and_ return a data frame.
+
+```{r}
+go_enrichment_results <-
+  WebGestaltR(
+    enrichMethod = "ORA",
+    organism = "hsapiens",
+    enrichDatabase = "geneontology_Biological_Process", # What categories we'd like enrichment results for
+    interestGene = upregulated_genes, # The vector of our genes of interest
+    interestGeneType = "ensembl_gene_id",
+    referenceGene = all_genes, # The vector of all the genes originally in our set
+    referenceGeneType = "ensembl_gene_id",
+    outputDirectory = results_dir, # Replace with the directory you would like your results saved to
+    projectName = "GSE37418_up_SHH_lfc2_fdr0_05" # Replace with a relevant output name for the HTML file
+  )
+```
+
+As noted in the messages from `WebGestaltR`, the results are in `results/Project_GSE37418_up_SHH_lfc2_fdr0_05`.
+The HTML report is at [`results/Project_GSE37418_up_SHH_lfc2_fdr0_05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html`](./results/Project_GSE37418_up_SHH_lfc2_fdr0_05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html).
+
+## Read in a results table
+
+Let's take a look at one of the results tables look like by reading it into R. 
+It is a `.txt` file, however its values are separated by tabs, making it a TSV; hence we can read it in with `readr::read_tsv()` function. 
+
+```{r}
+results_df <- readr::read_tsv(
+  file.path(
+    results_dir,
+    "Project_GSE37418_up_SHH_lfc2_fdr0_05",
+    "enrichment_results_GSE37418_up_SHH_lfc2_fdr0_05.txt"
+  )
+)
+```
+
+Let's print out a preview of it.   
+
+```{r}
+head(results_df)
```

Comment:
It's a bit difficult to tell whether or not there are indeed a lot of pathways associated with ribosomes and translation when one is using the rendered html of this notebook. Do you think we can include `head(results_df$description)` to give a better view of what's in this column?

I am not sure what's best here because I believe it is also important to be aware of and get a preview of the other columns in the object.

Reply:
My point for this isn't to dwell on the results too much but to just get a preview of whats there and also the point about genesets not being independent from each other. 

I could use `head(data.frame(results_df))` so the results are printed out in a longer format apparently it squishes tibbles but not data.frames upon rendering.

Reply:
I also feel (but could very well be wrong) that if someone has gotten this far in the notebook they are probably running it themselves so they can see and dig into the results a bit themselves. But you can still see the word `translation` is popped up. 🤷‍♀️

Reply:
All valid points. The word `translation` does pop up (I just personally had to do a little more digging to verify that `translation` and `ribosomes` popped up a lot like you stated below), so this change is not absolutely necessary.

If anything, I believe ` head(data.frame(results_df))` would resolve my original comment, but that may require you to explain why you are using `data.frame()` here as well. Therefore, it is completely up to you, whether or not to include this change or leave it as is.